// @aws-sdk/client-codeartifact/dist-es/commands/CopyPackageVersionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CopyPackageVersionsRequest, CopyPackageVersionsResult } from "../models/models_0";
import { deserializeAws_restJson1CopyPackageVersionsCommand, serializeAws_restJson1CopyPackageVersionsCommand, } from "../protocols/Aws_restJson1";
var CopyPackageVersionsCommand = (function (_super) {
    __extends(CopyPackageVersionsCommand, _super);
    function CopyPackageVersionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CopyPackageVersionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "CopyPackageVersionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CopyPackageVersionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CopyPackageVersionsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CopyPackageVersionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CopyPackageVersionsCommand(input, context);
    };
    CopyPackageVersionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CopyPackageVersionsCommand(output, context);
    };
    return CopyPackageVersionsCommand;
}($Command));
export { CopyPackageVersionsCommand };
