// @aws-sdk/client-codeartifact/dist-es/commands/CreateRepositoryCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateRepositoryRequest, CreateRepositoryResult } from "../models/models_0";
import { deserializeAws_restJson1CreateRepositoryCommand, serializeAws_restJson1CreateRepositoryCommand, } from "../protocols/Aws_restJson1";
var CreateRepositoryCommand = (function (_super) {
    __extends(CreateRepositoryCommand, _super);
    function CreateRepositoryCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateRepositoryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "CreateRepositoryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateRepositoryRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateRepositoryResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateRepositoryCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateRepositoryCommand(input, context);
    };
    CreateRepositoryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateRepositoryCommand(output, context);
    };
    return CreateRepositoryCommand;
}($Command));
export { CreateRepositoryCommand };
