// @aws-sdk/client-codeartifact/dist-es/commands/DeletePackageVersionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeletePackageVersionsRequest, DeletePackageVersionsResult } from "../models/models_0";
import { deserializeAws_restJson1DeletePackageVersionsCommand, serializeAws_restJson1DeletePackageVersionsCommand, } from "../protocols/Aws_restJson1";
var DeletePackageVersionsCommand = (function (_super) {
    __extends(DeletePackageVersionsCommand, _super);
    function DeletePackageVersionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeletePackageVersionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "DeletePackageVersionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeletePackageVersionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeletePackageVersionsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeletePackageVersionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeletePackageVersionsCommand(input, context);
    };
    DeletePackageVersionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeletePackageVersionsCommand(output, context);
    };
    return DeletePackageVersionsCommand;
}($Command));
export { DeletePackageVersionsCommand };
