// @aws-sdk/client-codeartifact/dist-es/commands/DeleteDomainPermissionsPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDomainPermissionsPolicyRequest, DeleteDomainPermissionsPolicyResult } from "../models/models_0";
import { deserializeAws_restJson1DeleteDomainPermissionsPolicyCommand, serializeAws_restJson1DeleteDomainPermissionsPolicyCommand, } from "../protocols/Aws_restJson1";
var DeleteDomainPermissionsPolicyCommand = (function (_super) {
    __extends(DeleteDomainPermissionsPolicyCommand, _super);
    function DeleteDomainPermissionsPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDomainPermissionsPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "DeleteDomainPermissionsPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDomainPermissionsPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteDomainPermissionsPolicyResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDomainPermissionsPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteDomainPermissionsPolicyCommand(input, context);
    };
    DeleteDomainPermissionsPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteDomainPermissionsPolicyCommand(output, context);
    };
    return DeleteDomainPermissionsPolicyCommand;
}($Command));
export { DeleteDomainPermissionsPolicyCommand };
