// @aws-sdk/client-codeartifact/dist-es/commands/DescribeRepositoryCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeRepositoryRequest, DescribeRepositoryResult } from "../models/models_0";
import { deserializeAws_restJson1DescribeRepositoryCommand, serializeAws_restJson1DescribeRepositoryCommand, } from "../protocols/Aws_restJson1";
var DescribeRepositoryCommand = (function (_super) {
    __extends(DescribeRepositoryCommand, _super);
    function DescribeRepositoryCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeRepositoryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "DescribeRepositoryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeRepositoryRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeRepositoryResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeRepositoryCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeRepositoryCommand(input, context);
    };
    DescribeRepositoryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeRepositoryCommand(output, context);
    };
    return DescribeRepositoryCommand;
}($Command));
export { DescribeRepositoryCommand };
