// @aws-sdk/client-codeartifact/dist-es/commands/DescribePackageCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribePackageRequest, DescribePackageResult } from "../models/models_0";
import { deserializeAws_restJson1DescribePackageCommand, serializeAws_restJson1DescribePackageCommand, } from "../protocols/Aws_restJson1";
var DescribePackageCommand = (function (_super) {
    __extends(DescribePackageCommand, _super);
    function DescribePackageCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribePackageCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "DescribePackageCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribePackageRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribePackageResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribePackageCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribePackageCommand(input, context);
    };
    DescribePackageCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribePackageCommand(output, context);
    };
    return DescribePackageCommand;
}($Command));
export { DescribePackageCommand };
