// @aws-sdk/client-codeartifact/dist-es/commands/DescribePackageVersionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribePackageVersionRequest, DescribePackageVersionResult } from "../models/models_0";
import { deserializeAws_restJson1DescribePackageVersionCommand, serializeAws_restJson1DescribePackageVersionCommand, } from "../protocols/Aws_restJson1";
var DescribePackageVersionCommand = (function (_super) {
    __extends(DescribePackageVersionCommand, _super);
    function DescribePackageVersionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribePackageVersionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "DescribePackageVersionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribePackageVersionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribePackageVersionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribePackageVersionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribePackageVersionCommand(input, context);
    };
    DescribePackageVersionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribePackageVersionCommand(output, context);
    };
    return DescribePackageVersionCommand;
}($Command));
export { DescribePackageVersionCommand };
