// @aws-sdk/client-codeartifact/dist-es/commands/DisposePackageVersionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisposePackageVersionsRequest, DisposePackageVersionsResult } from "../models/models_0";
import { deserializeAws_restJson1DisposePackageVersionsCommand, serializeAws_restJson1DisposePackageVersionsCommand, } from "../protocols/Aws_restJson1";
var DisposePackageVersionsCommand = (function (_super) {
    __extends(DisposePackageVersionsCommand, _super);
    function DisposePackageVersionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisposePackageVersionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "DisposePackageVersionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisposePackageVersionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DisposePackageVersionsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisposePackageVersionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DisposePackageVersionsCommand(input, context);
    };
    DisposePackageVersionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DisposePackageVersionsCommand(output, context);
    };
    return DisposePackageVersionsCommand;
}($Command));
export { DisposePackageVersionsCommand };
