// @aws-sdk/client-codeartifact/dist-es/commands/GetPackageVersionReadmeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetPackageVersionReadmeRequest, GetPackageVersionReadmeResult } from "../models/models_0";
import { deserializeAws_restJson1GetPackageVersionReadmeCommand, serializeAws_restJson1GetPackageVersionReadmeCommand, } from "../protocols/Aws_restJson1";
var GetPackageVersionReadmeCommand = (function (_super) {
    __extends(GetPackageVersionReadmeCommand, _super);
    function GetPackageVersionReadmeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetPackageVersionReadmeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "GetPackageVersionReadmeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetPackageVersionReadmeRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetPackageVersionReadmeResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetPackageVersionReadmeCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetPackageVersionReadmeCommand(input, context);
    };
    GetPackageVersionReadmeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetPackageVersionReadmeCommand(output, context);
    };
    return GetPackageVersionReadmeCommand;
}($Command));
export { GetPackageVersionReadmeCommand };
