// @aws-sdk/client-codeartifact/dist-es/commands/GetRepositoryPermissionsPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetRepositoryPermissionsPolicyRequest, GetRepositoryPermissionsPolicyResult } from "../models/models_0";
import { deserializeAws_restJson1GetRepositoryPermissionsPolicyCommand, serializeAws_restJson1GetRepositoryPermissionsPolicyCommand, } from "../protocols/Aws_restJson1";
var GetRepositoryPermissionsPolicyCommand = (function (_super) {
    __extends(GetRepositoryPermissionsPolicyCommand, _super);
    function GetRepositoryPermissionsPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetRepositoryPermissionsPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "GetRepositoryPermissionsPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetRepositoryPermissionsPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetRepositoryPermissionsPolicyResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetRepositoryPermissionsPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetRepositoryPermissionsPolicyCommand(input, context);
    };
    GetRepositoryPermissionsPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetRepositoryPermissionsPolicyCommand(output, context);
    };
    return GetRepositoryPermissionsPolicyCommand;
}($Command));
export { GetRepositoryPermissionsPolicyCommand };
