// @aws-sdk/client-codeartifact/dist-es/commands/ListPackagesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPackagesRequest, ListPackagesResult } from "../models/models_0";
import { deserializeAws_restJson1ListPackagesCommand, serializeAws_restJson1ListPackagesCommand, } from "../protocols/Aws_restJson1";
var ListPackagesCommand = (function (_super) {
    __extends(ListPackagesCommand, _super);
    function ListPackagesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListPackagesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "ListPackagesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPackagesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListPackagesResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPackagesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListPackagesCommand(input, context);
    };
    ListPackagesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListPackagesCommand(output, context);
    };
    return ListPackagesCommand;
}($Command));
export { ListPackagesCommand };
