// @aws-sdk/client-codeartifact/dist-es/commands/PutRepositoryPermissionsPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutRepositoryPermissionsPolicyRequest, PutRepositoryPermissionsPolicyResult } from "../models/models_0";
import { deserializeAws_restJson1PutRepositoryPermissionsPolicyCommand, serializeAws_restJson1PutRepositoryPermissionsPolicyCommand, } from "../protocols/Aws_restJson1";
var PutRepositoryPermissionsPolicyCommand = (function (_super) {
    __extends(PutRepositoryPermissionsPolicyCommand, _super);
    function PutRepositoryPermissionsPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutRepositoryPermissionsPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "PutRepositoryPermissionsPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutRepositoryPermissionsPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: PutRepositoryPermissionsPolicyResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutRepositoryPermissionsPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1PutRepositoryPermissionsPolicyCommand(input, context);
    };
    PutRepositoryPermissionsPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1PutRepositoryPermissionsPolicyCommand(output, context);
    };
    return PutRepositoryPermissionsPolicyCommand;
}($Command));
export { PutRepositoryPermissionsPolicyCommand };
