// @aws-sdk/client-codeartifact/dist-es/commands/UpdateRepositoryCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateRepositoryRequest, UpdateRepositoryResult } from "../models/models_0";
import { deserializeAws_restJson1UpdateRepositoryCommand, serializeAws_restJson1UpdateRepositoryCommand, } from "../protocols/Aws_restJson1";
var UpdateRepositoryCommand = (function (_super) {
    __extends(UpdateRepositoryCommand, _super);
    function UpdateRepositoryCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateRepositoryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeartifactClient";
        var commandName = "UpdateRepositoryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateRepositoryRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateRepositoryResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateRepositoryCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateRepositoryCommand(input, context);
    };
    UpdateRepositoryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateRepositoryCommand(output, context);
    };
    return UpdateRepositoryCommand;
}($Command));
export { UpdateRepositoryCommand };
